[BUG] `orchestrate.sh code-review` fails with "Argument list too long" with large diffs

## Description

`orchestrate.sh code-review` fails with "Argument list too long" during Round 2 (verification) when reviewing a PR with a moderately large diff. The diff content is embedded directly in shell arguments passed to `/usr/bin/timeout` and agent CLI commands, which exceeds the kernel's `ARG_MAX` limit.

## Steps to Reproduce

1. Have an open PR with >2000 lines of diff (tested with 3,396 lines / ~100KB)
2. Fix Bug 1 (JSON parse error) first so the pipeline reaches Round 1
3. Run: `orchestrate.sh code-review '{"target":"34","focus":["correctness","security"]}'`
4. Round 1 spawns agents successfully
5. Round 2 verification fails with "Argument list too long"

## Expected Behavior

The review pipeline should complete all 3 rounds (parallel fleet, verification, synthesis) regardless of diff size, up to reasonable limits.

## Actual Behavior

```
[INFO] review_run: Round 1 — parallel specialist fleet
[INFO] Spawning gemini agent (task: review-r1-security-reviewer-..., role: security-reviewer)
[INFO] Spawning claude-sonnet agent (task: review-r1-arch-reviewer-..., role: arch-reviewer)
[INFO] Spawning codex agent (task: review-r1-logic-reviewer-..., role: logic-reviewer)
[INFO] review_run: Round 1 complete
[INFO] review_run: Round 2 — verification
[WARN] review_run: codex verifier failed, falling back to claude-sonnet
[ERROR] Agent claude-sonnet failed with exit code 126 (role=code-reviewer, phase=review)
[ERROR] Error details: .../orchestrate.sh: line 7250: /usr/bin/timeout: Argument list too long
[WARN] review_run: verification failed entirely, using all findings as confirmed
[INFO] review_run: Round 3 — synthesis
[ERROR] Agent claude-sonnet failed with exit code 1 (role=code-reviewer, phase=review)
```

The pipeline degrades: Round 2 skips verification, Round 3 synthesis also fails, resulting in "No issues found" despite the diff having real issues.

## Environment

- OS: Ubuntu 24.04 (Linux 6.17.0-19-generic)
- Bash version: 5.2
- Claude Octopus version: 9.2.0
- Claude Code: v2.1.76
- Codex CLI: v0.114.0
- Gemini CLI: v0.33.1
- PR diff size: 3,396 lines (~100KB)
- Linux ARG_MAX: 2097152 bytes (2MB), but combined prompt + diff + context + environment exceeds this

## Logs

```
[ERROR] Error details: /home/steve/.claude/plugins/cache/nyldn-plugins/octo/9.2.0/scripts/orchestrate.sh: line 7250: /usr/bin/timeout: Argument list too long
```

## Additional Context

The issue is that the review prompt (which includes the full diff, review context, and instructions) is passed as a shell argument to `timeout` and then to the agent CLI. For medium-to-large PRs, this exceeds `ARG_MAX`.

Suggested fix: Write the prompt to a temp file and pipe it via stdin to agent CLIs, or pass the file path as an argument. Both `codex exec` and `gemini -p` support reading from stdin:

```bash
# Instead of:
timeout 180 codex exec --full-auto "$huge_prompt"

# Use:
echo "$huge_prompt" | timeout 180 codex exec --full-auto -
# Or:
echo "$huge_prompt" > "$tmpfile"
timeout 180 codex exec --full-auto < "$tmpfile"
```

--- a/scripts/orchestrate.sh
+++ b/scripts/orchestrate.sh
@@ -12580,18 +12580,12 @@ ${earned_skills_ctx}"
         local exit_code=0
         while true; do
             exit_code=0
-            if [[ "$agent_type" == gemini* ]]; then
-                if printf '%s' "$enhanced_prompt" | run_with_timeout "$TIMEOUT" "${cmd_array[@]}" 2> "$temp_errors" | tee "$raw_output" > "$temp_output"; then
-                    exit_code=0
-                else
-                    exit_code=$?
-                fi
+            # v9.2.1: All agents use stdin-based prompt delivery to avoid ARG_MAX limits (Issue #173)
+            # Previously only gemini used stdin; codex/claude passed prompt as CLI arg which fails on large diffs
+            if printf '%s' "$enhanced_prompt" | run_with_timeout "$TIMEOUT" "${cmd_array[@]}" 2> "$temp_errors" | tee "$raw_output" > "$temp_output"; then
+                exit_code=0
             else
-                if run_with_timeout "$TIMEOUT" "${cmd_array[@]}" "$enhanced_prompt" 2> "$temp_errors" | tee "$raw_output" > "$temp_output"; then
-                    exit_code=0
-                else
-                    exit_code=$?
-                fi
+                exit_code=$?
             fi
 
             # v8.16: Check if failure is auth-related and retryable
@@ -16391,10 +16385,9 @@ ${earned_skills_ctx}"
     # -p "" triggers headless mode; prompt content comes via stdin to avoid OS arg limits
     if [[ "$agent_type" == gemini* ]]; then
         cmd_array+=(-p "")
-        output=$(printf '%s' "$enhanced_prompt" | run_with_timeout "$timeout_secs" "${cmd_array[@]}" 2>"$temp_err")
-    else
-        output=$(run_with_timeout "$timeout_secs" "${cmd_array[@]}" "$enhanced_prompt" 2>"$temp_err")
     fi
+    # v9.2.1: All agents use stdin to avoid ARG_MAX "Argument list too long" on large diffs (Issue #173)
+    output=$(printf '%s' "$enhanced_prompt" | run_with_timeout "$timeout_secs" "${cmd_array[@]}" 2>"$temp_err")
     exit_code=$?
 
     # Check exit code and handle errors
@@ -17059,18 +17052,11 @@ ${enhanced_prompt}"
 
     while true; do
         exit_code=0
-        if [[ "$agent_type" == gemini* ]]; then
-            if printf '%s' "$enhanced_prompt" | run_with_timeout "$TIMEOUT" "${cmd_array[@]}" 2> "$temp_errors" | tee "$raw_output" > "$temp_output"; then
-                exit_code=0
-            else
-                exit_code=$?
-            fi
+        # v9.2.1: All agents use stdin to avoid ARG_MAX "Argument list too long" on large diffs (Issue #173)
+        if printf '%s' "$enhanced_prompt" | run_with_timeout "$TIMEOUT" "${cmd_array[@]}" 2> "$temp_errors" | tee "$raw_output" > "$temp_output"; then
+            exit_code=0
         else
-            if run_with_timeout "$TIMEOUT" "${cmd_array[@]}" "$enhanced_prompt" 2> "$temp_errors" | tee "$raw_output" > "$temp_output"; then
-                exit_code=0
-            else
-                exit_code=$?
-            fi
+            exit_code=$?
         fi
 
         if [[ $exit_code -ne 0 ]] && [[ $auth_attempt -lt $max_auth_retries ]]; then

